src/components/ButtonModal.tsx

```diff
@@ -0,0 +1,36 @@
+import { ReactNode, useRef } from "react";
+import { useMenuTrigger } from "react-aria";
+import { useMenuTriggerState } from "react-stately";
+import {
+  isIconButton,
+  isTextButton,
+  OverlayTrigger,
+  OverlayTriggerProps,
+} from "src/components/internal/OverlayTrigger";
+import { useTestIds } from "src/utils";
+import { ContextualModal } from "./internal/ContextualModal";
+
+export interface ContextualModalProps
+  extends Pick<OverlayTriggerProps, "trigger" | "placement" | "disabled" | "tooltip"> {
+  content: ReactNode;
+  title?: string;
+  // for storybook purposes
+  defaultOpen?: boolean;
```

Comment:
Suggestion: Maybe prefix this like `storybookDefaultOpen` so that engineers using the component, who may not see the in-source comment, easily know "ah yeah I don't need to worry about this".

Reply:
got it. ty!